src/game/SpellEffects.cpp

```diff
@@ -3700,15 +3714,19 @@ void Spell::DoSummon(SpellEffectIndex eff_idx)
 
     if (m_caster->GetTypeId() == TYPEID_PLAYER)
     {
-        // this enables popup window (pet dismiss, cancel)
         spawnCreature->SetUInt32Value(UNIT_FIELD_FLAGS, UNIT_FLAG_PVP_ATTACKABLE);
 
-        spawnCreature->GetCharmInfo()->SetPetNumber(pet_number, true);
+        CharmInfo* charmInfo = spawnCreature->GetCharmInfo();
+        if (m_caster->getClass() == CLASS_PRIEST)
```

Comment:
Atleast not in TBC from what I know. But if you want to keep it sync with others, DK Gargoyle for example should also start attacking on Summon. (Which is currently handled by a hack in TrinityCore)